pywwt/core.py

```diff
@@ -26,11 +27,17 @@ def _on_trait_change(self, changed):
         # its own, we only want to react to changes in traits that have the wwt
         # metadata attribute (which indicates the name of the corresponding WWT
         # setting).
-        wwt_name = self.trait_metadata(changed['name'], 'wwt')
+        wwt_name  = self.trait_metadata(changed['name'], 'wwt')
+        new_value = changed['new']
         if wwt_name is not None:
-            self._send_msg(event='setting_set',
-                           setting=wwt_name,
-                           value=changed['new'])
+            if hasattr(new_value, 'value') and hasattr(new_value, 'unit'):
```

Comment:
I think it would be cleaner here to do ``if isinstance(new_value, Quantity):`` where ``Quantity`` is the Astropy Quantity class. Maybe then you could even do:

```python
if isinstance(new_value, Quantity):
    new_value = new_value.value
```

then just have ``_send_msg`` once below.